### DISCUSSION TITLE
Build Failed after upgrading Maui.Markup package from 1.1.1 to 3.3.0

### DISCUSSION THREAD

**Discussion by gerhartz** [CommunityToolkit/Maui.Markup#248] [Q&A] [answered]
I upgraded our Maui project that was using CommunityToolkit.Maui.Markup 1.1.1 --> 3.3.0 and am now unable to build the project. I'm probably missing a step on my end, but I can't figure out what needs to change.

I'm getting a warning CS9057 and an Error CS1061. I imagine the build is failing because of the Error CS1061 but even when I open the file `TestSliders.cs` there is no IDE error that it can't find the `.TextTop()` method.

I'm not sure how to specify the compiler version of 4.7.0.0 anywhere. 

Any help is much appreciated!

Build error output:

`CSC: Warning CS9057 : The analyzer assembly '/Users/pgerhartz/.nuget/packages/communitytoolkit.maui.markup/3.3.0/analyzers/dotnet/cs/CommunityToolkit.Maui.Markup.SourceGenerators.dll' references version '4.7.0.0' of the compiler, which is newer than the currently running version '4.6.0.0'.
1>TestSliders.cs(39,18): Error CS1061 : 'Label' does not contain a definition for 'TextTop' and no accessible extension method 'TextTop' accepting a first argument of type 'Label' could be found (are you missing a using directive or an assembly reference?)
`

Here's some output of my environment:

`dotnet sdk check`

`.NET SDKs:
Version      Status                     
----------------------------------------
6.0.408      Patch 6.0.414 is available.
6.0.414      Up to date.                
7.0.203      Up to date.                
7.0.302      Patch 7.0.308 is available.
7.0.308      Up to date.                

Try out the newest .NET SDK features with .NET 8.0.100-rc.1.23463.5.

.NET Runtimes:
Name                          Version      Status                    
---------------------------------------------------------------------
Microsoft.AspNetCore.App      6.0.16       Patch 6.0.22 is available.
Microsoft.NETCore.App         6.0.16       Patch 6.0.22 is available.
Microsoft.AspNetCore.App      6.0.22       Up to date.               
Microsoft.NETCore.App         6.0.22       Up to date.               
Microsoft.AspNetCore.App      7.0.5        Patch 7.0.11 is available.
Microsoft.NETCore.App         7.0.5        Patch 7.0.11 is available.
Microsoft.AspNetCore.App      7.0.11       Up to date.               
Microsoft.NETCore.App         7.0.11       Up to date.   `

`dotnet --version`
7.0.308

`dotnet workload list`

Installed Workload Id      Manifest Version       Installation Source
---------------------------------------------------------------------
maui-tizen                 7.0.92/7.0.100         SDK 7.0.300        
maui-ios                   7.0.92/7.0.100         SDK 7.0.300        
maui-android               7.0.92/7.0.100         SDK 7.0.300        
maccatalyst                16.4.7099/7.0.100      SDK 7.0.300     

Some things I've tried that haven't seemed to make any difference...

1. Deleted all the `bin` and `obj` folders in my projects and reloaded project
2. Updated my Visual Studio with the latest .NET version
3. Added a using statement to MauiProject.cs for `using CommunityToolkit.Maui;`. We already have a global using statement for this so it did not do anything.
4. Cried a little bit & said a prayer. No luck there.

Thank you in advance for pointing me in the right direction!

**Comment by gerhartz** [comment]
Interesting - I removed all the usages of these and now its building:

`TextCenter()`
`TextCenterVertically()`
`TextTop()`
`TextEnd()`

No more warnings or errors. But it seems like these methods are still in the SDK to use 🤔

**Comment by TheCodeTraveler** [comment, marked as answer]
(See Issue #250)

I've experienced this bug too, but haven't figured out the fix yet. 

In my experience, it happens on macOS but not on Windows. 

However, it works in our CI builds. This leads me to assume it's a local environment configuration problem.  

I'll keep poking at it to see if I can figure out the culprit. 

In the meantime, check out our CI Pipeline environment configuration to see how to best configure your local environment: https://github.com/CommunityToolkit/Maui.Markup/blob/main/azure-pipelines.yml

## Workaround

An easy work-around is to downgrade to `CommunityToolkit.Maui.Markup v3.2.0`. The only new feature added to v3.3.0 is .NET Hot Reload support.

**Reply by gerhartz** [reply]
Downgrading to 3.2.0 works just fine. And yes I am on a Mac.

Thank you for the help!
